clang/test/Sema/attr-nonblocking-constraints.mm

```diff
@@ -0,0 +1,190 @@
+// RUN: %clang_cc1 -fsyntax-only -fblocks -fcxx-exceptions -verify %s
+// These are in a separate file because errors (e.g. incompatible attributes) currently prevent
```

Comment:
Are all of these specifically Objective-C++ tests? Because putting non-Objective C++ tests in an `.mm` file seems very weird to me. Also, Objective C++ tests should go in `test/SemaObjCXX` instead.

Reply:
Split the file.